lib/ood_core/job/adapters/lsf/batch.rb

```diff
@@ -0,0 +1,156 @@
+# Object used for simplified communication with a LSF batch server
+#
+# @note Used by Lsf adapter and not meant to be used directly.
+# @private
+class OodCore::Job::Adapters::Lsf::Batch
+  # TODO:
+  # attr_reader :cluster
+
+  # The path to the LSF client installation binaries
+  # @example For LSF 8.3
+  #   my_batch.bin.to_s #=> "/opt/lsf/8.3/linux2.6-glibc2.3-x86_64/bin"
+  # @return [Pathname] path to LSF binaries
+  attr_reader :bin
+
+  # The root exception class that all LSF-specific exceptions inherit
+  # from
+  class Error < StandardError; end
+
+  # @param cluster [#to_s] the cluster name
+  # @param bin [#to_s] path to LSF installation binaries
+  def initialize(bin: "")
+    # TODO: @cluster = cluster.to_s
+    @bin     = Pathname.new(bin.to_s)
+  end
+
+  # Get a list of hashes detailing each of the jobs on the batch server
+  # @param id [#to_s] the id of the job to check (if just checking one job)
+  # @param filters [Array<Symbol>] list of attributes to filter on
+  # @raise [Error] if `bjobs` command exited unsuccessfully
+  # @return [Array<Hash>] list of details for jobs
+  def get_jobs(id: "", filters: [])
+    id = id.to_s
+    #TODO: does filters make sense here? regardless this is a speed enhancement
+    # selected_fields = filters.empty? ? fields : (filters & fields)
+    #TODO: split into get_all_jobs, get_my_jobs, get_job?
+    args = bjobs_default_args
+
+    #TODO: get_jobs with a list of ids...
+    # get_jobs(ids = []) *(ids.map(&:to_s))
+    args << id unless id.empty?
+
+    parse_bjobs_output call("bjobs", *args)
+  end
+
+  def bjobs_default_args
+    %w( -u all -a -w -W )
+  end
+
+  # helper method
+  def parse_bjobs_output(response)
+    return [{}] if response =~ /No job found/ || response.nil?
+
+    lines = response.split("\n")
+    validate_bjobs_output_columns(lines.first.split)
+
+    lines.drop(1).map{ |job|
```

Comment:
> it was odd you used {} for a multi-line block instead of do end

I've actually seen arguments where in many cases this reads clearer than `do end`. There isn't a "hard fast" rule on "always use do end for multiple lines". I definitely prefer `{}` over `do end` if you will chain calls off of a map or reduce etc. In this case I could probably do either.